pkg/controller/ps/status.go

```diff
@@ -258,6 +270,24 @@ func (r *PerconaServerMySQLReconciler) isGRReady(ctx context.Context, cr *apiv1a
 	return true, nil
 }
 
+func (r *PerconaServerMySQLReconciler) isAsyncReady(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) (bool, error) {
+	pod, err := getOrcPod(ctx, r.Client, cr, 0)
+	if err != nil {
+		return false, err
+	}
+
+	p, err := orchestrator.ClusterPrimary(ctx, r.ClientCmd, pod, cr.ClusterHint())
+	if err != nil {
+		return false, err
+	}
+
+	if len(p.Replicas) < int(cr.Spec.MySQL.Size)-1 {
+		return false, nil
+	}
```

Comment:
rather than checking replicas count in `/api/master` output, I think it's better to check problems in `/api/cluster` so we can report which node has problems. for example:

```
bash-5.1$ curl -s localhost:3000/api/cluster/cluster1-mysql-0.cluster1-mysql.ps:3306/ | jq -r '.[] | "\(.Key.Hostname) \(.Problems)"'
cluster1-mysql-0.cluster1-mysql.ps []
cluster1-mysql-1.cluster1-mysql.ps []
cluster1-mysql-2.cluster1-mysql.ps ["not_replicating"]
```

Reply:
please also check orchestrator code potential values in `Problems` array

Reply:
[5a37239](https://github.com/percona/percona-server-mysql-operator/pull/667/commits/5a372396aea7de96c5dbaaecec267dcebce7542c)